ppl/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/parser/AstExpressionBuilderTest.java

```diff
@@ -0,0 +1,190 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.ppl.parser;
+
+import java.util.Collections;
+import org.junit.Test;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.agg;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.aggregate;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.and;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.doubleLiteral;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.equalTo;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.filter;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.function;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.in;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.intLiteral;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.not;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.or;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.project;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.relation;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.stringLiteral;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.plans.dsl.DSL.unresolvedAttr;
+
+public class AstExpressionBuilderTest extends AstBuilderTest{
+
+    @Test
+    public void testLogicalNotExpr() {
+        assertEqual("source=t not a=1",
+                filter(
+                        relation("t"),
+                        not(
+                                equalTo(unresolvedAttr("a"), intLiteral(1))
+                        )
+                ));
+    }
+
+    @Test
+    public void testLogicalOrExpr() {
+        assertEqual("source=t a=1 or b=2",
+                filter(
+                        relation("t"),
+                        or(
+                                equalTo(unresolvedAttr("a"), intLiteral(1)),
+                                equalTo(unresolvedAttr("b"), intLiteral(2))
+                        )
+                ));
+    }
+
+    @Test
+    public void testLogicalAndExpr() {
+        assertEqual("source=t a=1 and b=2",
+                filter(
+                        relation("t"),
+                        and(
+                                equalTo(unresolvedAttr("a"), intLiteral(1)),
+                                equalTo(unresolvedAttr("b"), intLiteral(2))
+                        )
+                ));
+    }
+
+    @Test
+    public void testLogicalAndExprWithoutKeyWordAnd() {
```

Comment:
should it be "Keyword" instead of "KeyWord"

Reply:
Will do, thanks!